.github/workflows/test-pr.yml

```diff
@@ -7,10 +7,50 @@ on:
       - completed
 
 jobs:
+  retrieve-build-url:
+    runs-on: ubuntu-latest
+    if: ${{ github.event.workflow_run.conclusion == 'success' }}
+    outputs:
+      url: ${{ steps.set-build-url.outputs.url }}
+    steps:
+      - name: 'Download build URL'
+        uses: actions/github-script@v6
+        with:
+          script: |
+            let allArtifacts = await github.rest.actions.listWorkflowRunArtifacts({
+               owner: context.repo.owner,
+               repo: context.repo.repo,
+               run_id: context.payload.workflow_run.id,
+            });
+            let matchArtifact = allArtifacts.data.artifacts.filter((artifact) => {
+              return artifact.name == "build_url"
+            })[0];
+            let download = await github.rest.actions.downloadArtifact({
+               owner: context.repo.owner,
+               repo: context.repo.repo,
+               artifact_id: matchArtifact.id,
+               archive_format: 'zip',
+            });
+            const fs = require('fs');
+            const path = require('path');
+            const temp = '${{ runner.temp }}/artifacts';
+            if (!fs.existsSync(temp)){
+              fs.mkdirSync(temp);
+            }
+            fs.writeFileSync(path.join(temp, 'build_url.zip'), Buffer.from(download.data));
+
+      - name: 'Setup build URL'
+        id: set-build-url
+        run: |
+          unzip "${{ runner.temp }}/artifacts/build_url.zip"
+          BUILD_URL=$(cat build_url)
+          echo "Build URL: ${BUILD_URL}"
+          echo "url=${BUILD_URL}" >> $GITHUB_OUTPUT
+
   test:
     uses: ./.github/workflows/test.yml
     secrets: inherit
-    if: ${{ github.event.workflow_run.conclusion == 'success' }}
+    needs: retrieve-build-url
```

Comment:
What for? Please expand commit message to describe why this is needed.

Reply:
Commit updated